test/cookie.test.js

```diff
@@ -847,3 +847,49 @@ test('should not decorate fastify, request and reply if no secret was provided',
     message: 'req.unsignCookie is not a function'
   })
 })
+
+test('dont add default @fastify/cookie hook-function to fastify if hook-option to false  ', (t) => {
+  t.plan(6)
+  const fastify = Fastify()
+  fastify.register(plugin, { hook: false })
+
+  for (const hook of ['preValidation', 'preHandler']) {
+    fastify.addHook(hook, (req, reply, done) => {
+      t.equal(req.cookies, null)
+      done()
+    })
+  }
+
+  fastify.addHook('preParsing', (req, reply, payload, done) => {
+    t.equal(req.cookies, null)
+    done()
+  })
+
+  fastify.get('/disable', (req, reply) => {
+    t.equal(req.cookies, null)
+    reply.send()
+  })
+
+  fastify.inject({
+    method: 'GET',
+    url: '/disable',
+    headers: {
+      cookie: 'bar=bar'
+    }
+  }, (err, res) => {
+    t.error(err)
+    t.equal(res.statusCode, 200)
+  })
+})
+
+test('result in an error if hook-option is set to an invalid value', (t) => {
+  t.plan(1)
+  const fastify = Fastify()
+
+  t.rejects(
+    () => fastify.register(plugin, { hook: true }),
+    {},
+    { skip: true },
```

Comment:
```suggestion
```

Reply:
I dont even know this notation :D. Why not simply the error as second param?